src/app/api/rest-client/route.ts

```diff
@@ -0,0 +1,41 @@
+'use server';
+
+import { RequestBody } from '@/types/interfaces';
+import { NextRequest, NextResponse } from 'next/server';
+import axios from 'axios';
+import getReadableErrorMessage from '@/shared/utils/get-readable-error-message';
+
+export async function POST(request: NextRequest) {
+  try {
+    const { method, url, headers = {}, data } = (await request.json()) as RequestBody;
+
+    const response = await axios({
+      method,
+      url,
+      headers: {
+        'User-Agent': 'REST-Client-App/1.0',
+        Accept: 'application/json',
+        ...headers,
+      },
+      data,
+    });
+
+    return NextResponse.json({
+      data: response.data,
+      status: response.status,
+      statusText: response.statusText,
+    });
```

Comment:
It hurts my eyes where you could add some deconstruction.
To be honest, because of ‘data’, it doesn't come out quite right.
This is just a suggestion, you don't have to listen to it.

```suggestion
     const { data: responseData, status, statusText } = await axios({
      method,
      url,
      headers: {
        'User-Agent': 'REST-Client-App/1.0',
        Accept: 'application/json',
        ...headers,
      },
      data,
    });

    return NextResponse.json({
      data: responseData,
      status,
      statusText,
    })
```

Reply:
I cannot let your eyes hurt, so I have applied you suggested changes 😄